### PR TITLE
add admin login with devise

### PR DESCRIPTION
**Pull request by abapolov**

### CODE REVIEW

**Comment by rakvium** [comment]
@abapolov could you please do `git pull --ff-only origin/master` on your branch?

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,274 @@
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = '78a07056f6977b0276a925033eb728381e043902f09084ea6511381ba526a6ebfde430bf42c26d00bbb743019d48ca2b2432ee4b8bbf8e61c628ea4d1c85bd82'
```
Comment (rakvium) [inline config/initializers/devise.rb]: @abapolov please remove this line, it's too long.

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  #
+  # Limiting the stretches to just one in testing will increase the performance of
+  # your test suite dramatically. However, it is STRONGLY RECOMMENDED to not use
+  # a value less than 10 in other environments. Note that, for bcrypt (the default
+  # algorithm), the cost increases exponentially with the number of stretches (e.g.
+  # a value of 20 is already extremely slow: approx. 60 seconds for 1 calculation).
+  config.stretches = Rails.env.test? ? 1 : 11
+
+  # Set up a pepper to generate the hashed password.
+  # config.pepper = '4e0999cb641e6807c3ffcf3586683ac5061c13c43b1f6684ec44ea45b7ca3a6e0c3b188815f7cac980e387515bef1eb6f9a8e12311ad0e222a3cbcdb5cd73300'
```
Comment (rakvium) [inline config/initializers/devise.rb]: @abapolov please remove this line, it's too long.

**Comment by rakvium** [comment]
@abapolov please add exclusion for `db/migrate/` folder and for `db/schema.rb` file to RuboCop config file.
```
rubocop .
Inspecting 49 files
........................C.......CC.CC.C..CC......

Offenses:

config/initializers/devise.rb:9:121: C: Line is too long. [154/120]
  # config.secret_key = '78a07056f6977b0276a925033eb728381e043902f09084ea6511381ba526a6ebfde430bf42c26d00bbb743019d48ca2b2432ee4b8bbf8e61c628ea4d1c85bd82'
                                                                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
config/initializers/devise.rb:111:121: C: Line is too long. [150/120]
  # config.pepper = '4e0999cb641e6807c3ffcf3586683ac5061c13c43b1f6684ec44ea45b7ca3a6e0c3b188815f7cac980e387515bef1eb6f9a8e12311ad0e222a3cbcdb5cd73300'
                                                                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/migrate/20161206130507_create_dispatchers.rb:1:1: C: Missing top-level class documentation comment.
class CreateDispatchers < ActiveRecord::Migration[5.0]
^^^^^
db/migrate/20161206115524_create_clients.rb:1:1: C: Missing top-level class documentation comment.
class CreateClients < ActiveRecord::Migration[5.0]
^^^^^
db/migrate/20161207125550_add_devise_to_admins.rb:1:1: C: Missing top-level class documentation comment.
class AddDeviseToAdmins < ActiveRecord::Migration[5.0]
^^^^^
db/migrate/20161207125550_add_devise_to_admins.rb:4:1: C: Extra empty line detected at block body beginning.
db/migrate/20161207125550_add_devise_to_admins.rb:33:1: C: Trailing whitespace detected.
db/migrate/20161206131603_create_drivers.rb:1:1: C: Missing top-level class documentation comment.
class CreateDrivers < ActiveRecord::Migration[5.0]
^^^^^
db/migrate/20161206131603_create_drivers.rb:8:34: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
      t.string :status, default: "not active"
                                 ^^^^^^^^^^^^
db/schema.rb:13:1: C: Block has too many lines. [43/25]
ActiveRecord::Schema.define(version: 20161207125550) do ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:13:38: C: Separate every 3 digits in the integer portion of a number with underscores(_).
ActiveRecord::Schema.define(version: 20161207125550) do
                                     ^^^^^^^^^^^^^^
db/schema.rb:14:1: C: Extra empty line detected at block body beginning.
db/schema.rb:16:20: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
  enable_extension "plpgsql"
                   ^^^^^^^^^
db/schema.rb:18:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
  create_table "admins", force: :cascade do |t|
               ^^^^^^^^
db/schema.rb:19:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "name",                               null: false
               ^^^^^^
db/schema.rb:20:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "email",                              null: false
               ^^^^^^^
db/schema.rb:21:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "password",                           null: false
               ^^^^^^^^^^
db/schema.rb:22:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "created_at",                         null: false
               ^^^^^^^^^^^^
db/schema.rb:23:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "updated_at",                         null: false
               ^^^^^^^^^^^^
db/schema.rb:24:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "reset_password_token"
               ^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:25:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "reset_password_sent_at"
               ^^^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:26:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "remember_created_at"
               ^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:27:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.integer  "sign_in_count",          default: 0, null: false
               ^^^^^^^^^^^^^^^
db/schema.rb:27:32: C: Unnecessary spacing detected.
    t.integer  "sign_in_count",          default: 0, null: false
                               ^^^^^^^^^
db/schema.rb:28:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "current_sign_in_at"
               ^^^^^^^^^^^^^^^^^^^^
db/schema.rb:29:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "last_sign_in_at"
               ^^^^^^^^^^^^^^^^^
db/schema.rb:30:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.inet     "current_sign_in_ip"
               ^^^^^^^^^^^^^^^^^^^^
db/schema.rb:31:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.inet     "last_sign_in_ip"
               ^^^^^^^^^^^^^^^^^
db/schema.rb:32:14: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["email"], name: "index_admins_on_email", unique: true, using: :btree
             ^^^^^^^
db/schema.rb:32:30: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["email"], name: "index_admins_on_email", unique: true, using: :btree
                             ^^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:33:14: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["reset_password_token"], name: "index_admins_on_reset_password_token", unique: true, using: :btree
             ^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:33:45: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["reset_password_token"], name: "index_admins_on_reset_password_token", unique: true, using: :btree
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:36:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
  create_table "clients", force: :cascade do |t|
               ^^^^^^^^^
db/schema.rb:37:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "email"
               ^^^^^^^
db/schema.rb:38:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "phone",      null: false
               ^^^^^^^
db/schema.rb:39:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "created_at", null: false
               ^^^^^^^^^^^^
db/schema.rb:40:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "updated_at", null: false
               ^^^^^^^^^^^^
db/schema.rb:41:14: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["phone"], name: "index_clients_on_phone", unique: true, using: :btree
             ^^^^^^^
db/schema.rb:41:30: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["phone"], name: "index_clients_on_phone", unique: true, using: :btree
                             ^^^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:44:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
  create_table "dispatchers", force: :cascade do |t|
               ^^^^^^^^^^^^^
db/schema.rb:45:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "email",      null: false
               ^^^^^^^
db/schema.rb:46:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "password",   null: false
               ^^^^^^^^^^
db/schema.rb:47:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "name",       null: false
               ^^^^^^
db/schema.rb:48:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "created_at", null: false
               ^^^^^^^^^^^^
db/schema.rb:49:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "updated_at", null: false
               ^^^^^^^^^^^^
db/schema.rb:50:14: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["email"], name: "index_dispatchers_on_email", unique: true, using: :btree
             ^^^^^^^
db/schema.rb:50:30: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["email"], name: "index_dispatchers_on_email", unique: true, using: :btree
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:53:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
  create_table "drivers", force: :cascade do |t|
               ^^^^^^^^^
db/schema.rb:54:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "name",                              null: false
               ^^^^^^
db/schema.rb:55:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "phone",                             null: false
               ^^^^^^^
db/schema.rb:56:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "pass",                              null: false
               ^^^^^^
db/schema.rb:57:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "auto",                              null: false
               ^^^^^^
db/schema.rb:58:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "status",     default: "not active"
               ^^^^^^^^
db/schema.rb:58:25: C: Unnecessary spacing detected.
    t.string   "status",     default: "not active"
                        ^^^^
db/schema.rb:58:39: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.string   "status",     default: "not active"
                                      ^^^^^^^^^^^^
db/schema.rb:59:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "created_at",                        null: false
               ^^^^^^^^^^^^
db/schema.rb:60:16: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.datetime "updated_at",                        null: false
               ^^^^^^^^^^^^
db/schema.rb:61:14: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["phone"], name: "index_drivers_on_phone", unique: true, using: :btree
             ^^^^^^^
db/schema.rb:61:30: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    t.index ["phone"], name: "index_drivers_on_phone", unique: true, using: :btree
                             ^^^^^^^^^^^^^^^^^^^^^^^^
db/schema.rb:63:1: C: Extra empty line detected at block body end.
app/models/admin.rb:1:1: C: Missing top-level class documentation comment.
class Admin < ApplicationRecord
^^^^^
app/models/dispatcher.rb:1:1: C: Missing top-level class documentation comment.
class Dispatcher < ApplicationRecord
^^^^^

49 files inspected, 62 offenses detected
```

File: .rubocop.yml [hunk 3]
```diff
@@ -12,4 +12,6 @@ Metrics/ClassLength:
 
 Lint/AssignmentInCondition:
     Exclude: 
-      - 'bin/spring'
\ No newline at end of file
+      - 'bin/spring'
+      - 'db/migrate'
```
Comment (rakvium) [inline .rubocop.yml]: This is a wrong place for this folder. There are still `60 offenses`.

File: .rubocop.yml [hunk 4]
```diff
@@ -12,4 +12,6 @@ Metrics/ClassLength:
...
+      - 'db/schema.rb'
```
Comment (rakvium) [inline .rubocop.yml]: @abapolov this is a wrong place for this file. There are still `60 offenses`.
Please run `rubocop` before next commit :wink:

File: app/models/admin.rb [hunk 5]
```diff
@@ -1,3 +1,4 @@
+# class admin
```
Comment (rakvium) [inline app/models/admin.rb]: @abapolov can you please be more verbose in comments? `model for admin users` or something like this, at least.

Comment (EvgeniiK) [inline app/models/admin.rb]: Better to say that this model has CRUD for all models in the system.

**Pull request merged**